### PR TITLE
core: Deprecate/remove StartTestClient in favour of StartTestCoreWithApp

### LINKED ISSUE #1365: core: Deprecate/remove `StartTestClient` in favour of `StartTestCoreWithApp`
Replace all uses of `StartTestClient` with `StartTestCoreWithApp` in the codebase.

Based on merging of #1364

### PR DESCRIPTION
**Pull request by dougEfresh**
Fixes #1365

## Overview

Replace StartTestClient with StartTestCoreWithApp within tests

### CODE REVIEW

File: core/testing.go [hunk 1]
```diff
@@ -52,26 +52,16 @@ func CreateKVStore(retainBlocks int64) *kvstore.Application {
...
+func getFreePort() (port int, err error) {
+	var a *net.TCPAddr
+	if a, err = net.ResolveTCPAddr("tcp", "localhost:0"); err == nil {
+		var l *net.TCPListener
+		if l, err = net.ListenTCP("tcp", a); err == nil {
+			defer l.Close()
+			return l.Addr().(*net.TCPAddr).Port, nil
+		}
+	}
+	return
```
Comment (Wondertan) [inline core/testing.go:64]: Doesn't the zero port allocate the free one on the kernel? Why do we need to do it manually?

Comment (dougEfresh) [reply core/testing.go:64]: Later in the code there is a [GetEndpoint](https://github.com/celestiaorg/celestia-node/blob/main/core/testing.go#L116-L128) which is used to build the test client.

```
func GetEndpoint(cfg *config.Config) (string, error) {
	url, err := url.Parse(cfg.RPC.ListenAddress)
	if err != nil {
		return "", err
	}
	host, _, err := net.SplitHostPort(url.Host)
	if err != nil {
		return "", err
	}
	return fmt.Sprintf("%s:%s", host, url.Port()), nil
}
```

So I need to know which port the kernel gave the test in order to build the  client.  
I do this [here](https://github.com/celestiaorg/celestia-node/pull/1400/files#diff-9750405595d72572f9a78b11e628a3cf4331798bb84dcd88746e61a8ba215657L91)
```
	// get a random port for running test in parallel
	freePort, err := getFreePort()
	require.NoError(t, err)
	tmNode.Config().RPC.ListenAddress = fmt.Sprintf("tcp://127.0.0.1:%d", freePort)
	freePort, err = getFreePort()
	require.NoError(t, err)
	tmNode.Config().P2P.ListenAddress = fmt.Sprintf("tcp://0.0.0.0:%d", freePort)

```

File: core/client_test.go [hunk 2]
```diff
@@ -10,20 +10,20 @@ import (
 )
 
 func TestRemoteClient_Status(t *testing.T) {
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
```
Comment (dougEfresh) [inline core/client_test.go:13]: I increased the timeout because I saw  random timeouts when running tests locally

File: core/testing.go (hunk 1 above)
Comment (Wondertan) [reply core/testing.go:64]: Ok, I see. So we need RPC to get a port and create a client. Why do we need to get a free port for a P2P listen-to address, though? @dougEfresh

Comment (Wondertan) [reply core/testing.go:64]: Yeah, so let's just set it to 0

Comment (Wondertan) [reply core/testing.go:64]: and for RPC we can use a freePort

Comment (Wondertan) [reply core/testing.go:64]: But actually, this is a very minor thing to worry about, and using getFreePort is not that bad.
I just don't really like doing what kernel already provides, but its fine in this case

**Review by Wondertan — APPROVED** [review]

File: core/testing.go (hunk 1 above)
Comment (dougEfresh) [reply core/testing.go:64]: Ah, I see what you mean. Minor thing yes, but I changed p2p port to use port 0 and tests pass.

**Review by vgonkivs — APPROVED** [review]
LGTM

**Review by renaynay — APPROVED** [review]
Thanks for the contribution @dougEfresh :)

**Review by distractedm1nd — APPROVED** [review]
🫡

**Pull request merged**